src/AbstractDelegatingDecorator.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+/*
+ * This file is part of the webmozart/key-value-store package.
+ *
+ * (c) Bernhard Schussek <bschussek@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Webmozart\KeyValueStore;
+
+use Webmozart\KeyValueStore\Api\KeyValueStore;
+
+/**
+ * A delegating decorator delegate each call of a KeyValueStore method
+ * to the internal store.
+ *
+ * It is used by decorators that need to override only a few specific
+ * methods (such as SortableDecorator or CountableDecorator).
+ *
+ * @since  1.0
+ *
+ * @author Bernhard Schussek <bschussek@gmail.com>
+ * @author Titouan Galopin <galopintitouan@gmail.com>
+ */
+class AbstractDelegatingDecorator implements KeyValueStore
```

Comment:
A decorator delegates to the decorated class by definition, so I think you can simplify the name to `AbstractDecorator`.